src/pages/index.js

```diff
@@ -5,52 +5,281 @@ import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
 import PopupWithImage from "../components/PopupWithImages.js";
 import Popup from "../components/Popup.js";
+import Api from "../components/Api.js";
 import UserInfo from "../components/UserInfo.js";
+import ConfirmPopup from "../components/ConfirmPopup.js";
 import {
   initialCards,
   profileForm,
   newCardForm,
+  avatarEditForm,
   profileFormValidator,
   newCardFormValidator,
+  // avatarFormValidator,
   validationConfig,
   profileAddButton,
   addCardModal,
   editProfileModal,
   profileTitle,
   profileDescription,
   profileEditForm,
+  userInfo,
 } from "../utils/constants.js";
 
+const avatarFormValidator = new FormValidator(validationConfig, avatarEditForm);
+avatarFormValidator.enableValidation();
+
 const profileEditButton = document.querySelector("#profile-edit-button");
-// //
-// /*new code p5*/
+const avatarImage = document.querySelector(".edit_icon");
+const submitButtonSelector = document.querySelector(".modal__button");
 const addNewCardButton = document.querySelector(".profile__add-button");
 const profileTitleInput = document.querySelector("#profile-title-input");
 const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
 const cardListEl = document.querySelector(".cards__list");
-// export const profileForm = document.querySelector(
-//   "#profile-edit-modal .modal__form"
-// );
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  jobSelector: ".profile__description",
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "acb9f423-9227-4882-a1e0-a6492f6bb248",
+    "Content-Type": "application/json",
+  },
+});
+
+let section;
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cardData]) => {
+    userInfo.setUserInfo(userData);
+    section = new Section(
+      {
+        items: cardData,
+        renderer: (cardData) => {
+          renderCard(cardData);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+let user;
+
+// avatar edit modal
+
+const avatarChangeModal = new PopupWithForm(
+  "#avatar-modal",
+  handleAvatarChangeSubmit
+);
+avatarChangeModal.setEventListeners();
+
+// ---------------------------------------
+const avatarSubmitButton = document.querySelector("#avatar-submit-button");
+
+function handleAvatarChangeSubmit(inputValues) {
+  // event.preventDefault();
+
+  // const avatarForm = event.currentTarget;
+  const avatarUrl = inputValues.avatar;
+  console.log("avatarUrl:", avatarUrl);
+  // const avatarUrl = event.target.avatar.value;
+
+  if (avatarUrl) {
+    avatarSubmitButton.textContent = "Saving...";
+    api
+
+      .updateAvatar(avatarUrl)
+      .then((updatedUser) => {
+        userInfo.setUserInfo(updatedUser);
+        avatarChangeModal.close();
+        this._popupForm.reset();
```

Comment:
`_popupForm`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.


**Please, correct it everywhere in the code**